nodejs/scripts/jsonix.js

```diff
@@ -30,7 +30,7 @@ var _jsonix_factory = function (_jsonix_xmldom, _jsonix_xmlhttprequest, _jsonix_
 
 			// REWORK
 			// Node.js
-			sourceIsEvt = typeof window !== 'undefined' && window !== null && typeof window.Event === "function" && source instanceof window.Event;
+			var sourceIsEvt = typeof window !== 'undefined' && window !== null && typeof window.Event === "function" && source instanceof window.Event;

```

Comment:
What error occurred that prompted you to make this fix?  What are the implications of declaring this variable?  (Is the introduction of scope as opposed to it automatically being global causing an issue?)

Why did this commit in the weird other js dir do basically the opposite of what you're trying to do now?  https://github.com/mitre/jsonix/commit/8de7f1c7f259845020c15c1a5580f7e7e85ae387

This change was also explicitly made in this commit in this same file: https://github.com/mitre/jsonix/commit/e6e58e66614a5c7154a1f3456fd1a1ad89dceb0a#diff-b77f70d3beb1864de2c1fb848006899f2f2a87b33e72c43980065cef33db3564

Reply:
Strict mode is detecting a reference to an undeclared variable and is throwing this (it references the same error as the MDN page you linked):

<img width="497" alt="Screenshot 2023-08-23 at 8 18 56 AM" src="https://github.com/mitre/jsonix/assets/106095667/97564bd0-03ca-4939-ad87-34954f9ce857">

I'm not sure why the opposite was performed in the original repo, but I see no reason why restricting its scope is detrimental given that it's only ever used in that one instance (albeit I don't know the implications of why it even is global to begin with).